lib/IRGen/GenCall.cpp

```diff
@@ -4037,6 +4061,23 @@ static void externalizeArguments(IRGenFunction &IGF, const Callee &callee,
       continue;
     }
 
+    bool passIndirectToDirect = paramInfo.isIndirectInGuaranteed() && paramType.isSensitive();
+    if (passIndirectToDirect) {
+      llvm::Value *ptr = in.claimNext();
+
+      if (AI.getKind() == clang::CodeGen::ABIArgInfo::Indirect) {
```

Comment:
This is the case where the SIL claimed the C function took it `@in_guaranteed` (the callee preserves the content of the memory pointed to).

We pass the pointer to a C function that that lowers that argument indirect. It depends on the platform ABI whether the callee may modify that memory or not.

You need to create a temporary copy unless there would be some other guarantee in SIL that the SIL address passed to the C function call is not used after the call.